botocore.exceptions.HTTPClientError should be retryable.

Sometimes, [Modin's](https://github.com/modin-project/modin/) github CI test case `modin/pandas/test/test_io.py::TestFwf::test_read_fwf_s3[storage_options2]` fails because s3fs gets the boto error `HTTPClientError('An HTTP Client raised an unhandled exception: [Errno 104] Connection reset by peer')`. For example, see [this run](https://github.com/modin-project/modin/runs/5410388785?check_suite_focus=true), but ignore the `'NoneType' object is not subscriptable`: that is a separate bug in `s3fs.errors.translate_boto_error`. Scroll to the error just above that one. It has stack trace:

<details>
<summary>Show stack trace</summary>

```
self = <modin.pandas.test.test_io.TestFwf object at 0x7fb9f14a01c0>
storage_options = {'key': '123', 'secret': '123'}

    @pytest.mark.parametrize(
        "storage_options",
        [{"anon": False}, {"anon": True}, {"key": "123", "secret": "123"}, None],
    )
    def test_read_fwf_s3(self, storage_options):
>       eval_io(
            fn_name="read_fwf",
            filepath_or_buffer="s3://modin-datasets/testing/test_data.fwf",
            storage_options=storage_options,
        )

modin/pandas/test/test_io.py:2132: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
modin/pandas/test/utils.py:814: in eval_io
    call_eval_general()
modin/pandas/test/utils.py:797: in call_eval_general
    eval_general(
modin/pandas/test/utils.py:746: in eval_general
    values = execute_callable(
modin/pandas/test/utils.py:725: in execute_callable
    md_result = fn(modin_df, **md_kwargs)
modin/pandas/test/utils.py:791: in applyier
    result = getattr(module, fn_name)(*args, **kwargs)
modin/pandas/io.py:496: in read_fwf
    pd_obj = FactoryDispatcher.read_fwf(**target_kwargs)
modin/core/execution/dispatching/factories/dispatcher.py:257: in read_fwf
    return cls.__factory._read_fwf(**kwargs)
modin/core/execution/dispatching/factories/factories.py:337: in _read_fwf
    return cls.io_cls.read_fwf(**kwargs)
modin/core/io/io.py:712: in read_fwf
    pd_obj = pandas.read_fwf(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/util/_decorators.py:311: in wrapper
    return func(*args, **kwargs)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py:876: in read_fwf
    return _read(filepath_or_buffer, kwds)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py:575: in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py:933: in __init__
    self._engine = self._make_engine(f, self.engine)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/parsers/readers.py:1217: in _make_engine
    self.handles = get_handle(  # type: ignore[call-overload]
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/common.py:670: in get_handle
    ioargs = _get_filepath_or_buffer(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/pandas/io/common.py:396: in _get_filepath_or_buffer
    file_obj = fsspec.open(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/core.py:140: in open
    out = self.__enter__()
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/core.py:103: in __enter__
    f = self.fs.open(self.path, mode=mode)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/spec.py:1009: in open
    f = self._open(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/s3fs/core.py:573: in _open
    return S3File(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/s3fs/core.py:1834: in __init__
    super().__init__(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/spec.py:1351: in __init__
    self.size = self.details["size"]
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/spec.py:1364: in details
    self._details = self.fs.info(self.path)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/asyn.py:85: in wrapper
    return sync(self.loop, func, *args, **kwargs)
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/asyn.py:65: in sync
    raise return_result
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/fsspec/asyn.py:25: in _runner
    result[0] = await coro
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/s3fs/core.py:1065: in _info
    out = await self._call_s3(
/usr/share/miniconda/envs/modin/lib/python3.8/site-packages/s3fs/core.py:301: in _call_s3
    err = translate_boto_error(err)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

error = HTTPClientError('An HTTP Client raised an unhandled exception: [Errno 104] Connection reset by peer')
message = None, set_cause = True, args = (), kwargs = {}

    def translate_boto_error(error, message=None, set_cause=True, *args, **kwargs):
        """Convert a ClientError exception into a Python one.
    
        Parameters
        ----------
    
        error : botocore.exceptions.ClientError
            The exception returned by the boto API.
        message : str
            An error message to use for the returned exception. If not given, the
            error message returned by the server is used instead.
        set_cause : bool
            Whether to set the __cause__ attribute to the previous exception if the
            exception is translated.
        *args, **kwargs :
            Additional arguments to pass to the exception constructor, after the
            error message. Useful for passing the filename arguments to ``IOError``.
    
        Returns
        -------
    
        An instantiated exception ready to be thrown. If the error code isn't
        recognized, an IOError with the original error message is returned.
        """
        if not hasattr(error, "response"):
            # non-http error:
            return error
>       code = error.response["Error"].get("Code")
E       TypeError: 'NoneType' object is not subscriptable

```
</details>

It looks like no HTTPClientError is included in [`S3_RETRYABLE_ERRORS`](https://github.com/fsspec/s3fs/blob/79191aca2e545175b85fb2f693ac1c0b254d4334/s3fs/core.py#L52).  I think that at least this kind of HTTPClientError should be retryable.

I found #149, which reports a similar issue in Python 2.

--- a/s3fs/core.py
+++ b/s3fs/core.py
@@ -19,7 +19,7 @@ import aiobotocore
 import botocore
 import aiobotocore.session
 from aiobotocore.config import AioConfig
-from botocore.exceptions import ClientError, ParamValidationError
+from botocore.exceptions import ClientError, HTTPClientError, ParamValidationError
 
 from s3fs.errors import translate_boto_error
 from s3fs.utils import S3BucketRegionCache, ParamKwargsHelper, _get_brange, FileExpired
@@ -46,7 +46,7 @@ if "S3FS_LOGGING_LEVEL" in os.environ:
 
 
 MANAGED_COPY_THRESHOLD = 5 * 2**30
-S3_RETRYABLE_ERRORS = (socket.timeout, IncompleteRead)
+S3_RETRYABLE_ERRORS = (socket.timeout, HTTPClientError, IncompleteRead)
 
 if ClientPayloadError is not None:
     S3_RETRYABLE_ERRORS += (ClientPayloadError,)

--- a/s3fs/errors.py
+++ b/s3fs/errors.py
@@ -136,14 +136,15 @@ def translate_boto_error(error, message=None, set_cause=True, *args, **kwargs):
     An instantiated exception ready to be thrown. If the error code isn't
     recognized, an IOError with the original error message is returned.
     """
-    if not hasattr(error, "response"):
-        # non-http error:
+    error_response = getattr(error, "response", None)
+    if error_response is None:
+        # non-http error, or response is None:
         return error
-    code = error.response["Error"].get("Code")
+    code = error_response["Error"].get("Code")
     constructor = ERROR_CODE_TO_EXCEPTION.get(code)
     if constructor:
         if not message:
-            message = error.response["Error"].get("Message", str(error))
+            message = error_response["Error"].get("Message", str(error))
         custom_exc = constructor(message, *args, **kwargs)
     else:
         # No match found, wrap this in an IOError with the appropriate message.

